code/game/gamemodes/objectives/stash_component.dm

```diff
@@ -1,57 +1,64 @@
 /datum/component/stash
 	dupe_mode = COMPONENT_DUPE_ALLOWED
 	var/atom/movable/stash_item
-	var/datum/mind/stash_owner
+	var/list/stash_minds
```

Comment:
I made change from your version to mix it with my client image system. 
Please check: https://github.com/BeeStation/BeeStation-Hornet/pull/10789/commits/493e0dbd313645a1ec073256dceda9fe50df0597


* my client image system can keep client images to minds instead of mob based. This makes that you don't have to refresh mob based client images every time mind is changed or something. This is why I nuked `transfer_mind()` here.
Even if they aghost or got brain-transfered, they keep this, and you don't have to send a signal here.
* `add_membership()` and `remove_membership() is likely necessary here because someone can be removed, or someone can be added to the party.
Admin might want to add a new member to a stash when they have to fix or do some admeme.
  * This can be easily expanded to allow non-stash owners see this (i.e. Detective)

The proc `owner_deleted()` in the original code calls `create_owner_icon()` everytime when alternative appearance should be updated, but my client image system can simplify this much.

Reply:
https://github.com/BeeStation/BeeStation-Hornet/pull/10789/commits/31e665a770c4511a4c543c3b82f27205e966bdc5

Original stash code change example if this is too much